### PR TITLE
Add R checks

### PR DESCRIPTION
**Pull request by rossellabelleli**
#11 Progresses issue

### CODE REVIEW

**Review by maximo1311 — COMMENTED** [review]
@rossellabelleli thanks for adding these checks. Let's get @gravesti to review it.

**Comment by gravesti** [comment]
@rossellabelleli @maximo1311 I've added the github actions/workflows for the automatic checking of the package and set up the files for adding tests. I also fixed a few things for R CMD check.

@rossellabelleli As you are the requestor of this PR, it's not possible to add you as a reviewer.

File: .Rbuildignore
Comment (gravesti) [inline .Rbuildignore:1]: Add the files to the "ignore" file so that R doesn't look at them when building the package

File: .github/workflows/check.yaml
Comment (gravesti) [inline .github/workflows/check.yaml:1]: Lots of automated tests here. We might remove some if they aren't helpful

File: .github/workflows/docs.yaml
Comment (gravesti) [inline .github/workflows/docs.yaml:1]: This builds the pkgdown site and puts it in a separate branch. We probably will need to update the settings for that.

File: .github/workflows/release.yaml
Comment (gravesti) [inline .github/workflows/release.yaml:1]: This one should publish the package. It won't run for now.

File: DESCRIPTION
Comment (gravesti) [inline DESCRIPTION:1]: Update for testthat, roxygen and missing dependencies

**Review by gravesti — APPROVED** [review]

**Review by maximo1311 — APPROVED** [review]

**Pull request merged**